Packs/PATHelpdeskAdvanced/Integrations/PATHelpdeskAdvanced/PATHelpdeskAdvanced.py

```diff
@@ -0,0 +1,1044 @@
+        elif isinstance(item, list):
+            # mypy is confused by the type union
+            return [filter_fields(list_item) for list_item in item]  # type:ignore[misc]
+        raise TypeError(f"cannot filter {type(item)}, expected dict|list[dict]")
+
+    if field_replacements:
+        output = replace_fields(output)
+
+    if fields is not None:
+        output = filter_fields(output)
+
+    return tableToMarkdown(
+        name=title,
+        t=output,
+        headerTransform=pascalToSpace,
+        sort_headers=False,
+        **kwargs,
+    )
+
+
+def create_ticket_command(client: Client, args: dict) -> CommandResults:
+    raw_response = client.create_ticket(**args)
+    response = convert_response_dates(raw_response)
+
+    response_for_human_readable = response.copy()["data"]
+
+    if not response_for_human_readable.get(SOLUTION.hda_name):
+        # do not show empty or missing `Solution` value
+        response_for_human_readable.pop(SOLUTION.hda_name, None)
+
+    return CommandResults(
+        outputs_prefix=f"{VENDOR}.Ticket",
+        outputs_key_field=ID.hda_name,
+        outputs=response,
+        raw_response=raw_response,
+        readable_output=pat_table_to_markdown(
+            title="Ticket Created",
+            output=response_for_human_readable,
+            fields=(
+                TICKET_ID,
+                OBJECT_DESCTIPTION,
+                OBJECT_ENTITY,
+                SOLUTION,
+                TICKET_CLASSIFICATION_ID,
+                IS_NEW,
+                EXPIRATION_DATE,
+                FIRST_UPDATE_USER_ID,
+                OWNER_USER_ID,
+                DATE,
+                ASSIGNED_USER_ID,
+            ),
+        ),
+    )
+
+
+def list_tickets_command(client: Client, args: dict) -> CommandResults:
+    raw_response = client.list_tickets(**args)
+    response = convert_response_dates(raw_response)
+
+    return CommandResults(
+        outputs=response["data"],
+        raw_response=raw_response,
+        outputs_prefix=f"{VENDOR}.Ticket",
+        outputs_key_field=ID.hda_name,
+        readable_output=pat_table_to_markdown(
+            title="Tickets",
+            output=response["data"],
+            fields=(
+                TICKET_ID,  # Replaced from ID
+                SUBJECT,
+                SOLUTION,
+                DATE,
+                SERVICE_ID,
+                PROBLEM,
+                CONTACT_ID,
+                OWNER_USER_ID,
+                ACCOUNT_ID,
+            ),
+            field_replacements={ID: TICKET_ID},
+        ),
+    )
+
+
+def add_ticket_attachment_command(client: Client, args: dict) -> CommandResults:
+    entry_ids = argToList(args.pop("entry_id", ()))
+    ticket_id = args["ticket_id"]
+    response = client.add_ticket_attachment(entry_ids, **args)
+    return CommandResults(
+        readable_output=f"Added Attachment ID {response['data']['attachmentID']} to ticket ID {ticket_id}"
+        if len(entry_ids) == 1
```

Comment:
again, maybe we can consider adding one a attachment at time to make this simpler.

Reply:
discussed, will keep as is.